cpp/CMakeLists.txt

```diff
@@ -419,6 +419,7 @@ add_library(
   src/filling/fill.cu
   src/filling/repeat.cu
   src/filling/sequence.cu
+  src/filter/filter.cu
```

Comment:
This should probably be in `copying/` or `stream_compaction/`. We have other filter-like algorithms there. “Apply boolean mask” might be the closest.